src/tpl.rs

```diff
@@ -14,30 +14,41 @@ enum FileType {
 struct Tpl {
     type_: FileType,
     ntextures: u32,
-    texdesc: Vec<u8>,
+    texdesc: *mut libc::c_void,
     tpl_file: ffi::FHANDLE,
 }
 
 impl Tpl {
-    fn new() -> Self {
-        Self {
-            type_: FileType::Disc,
+    pub fn open_from_memory(memory: &mut [u8]) -> Self {
+        let mut tpl = Self {
+            type_: FileType::Memory,
             ntextures: 0,
-            texdesc: Vec::new(),
+            texdesc: 0 as _,
             tpl_file: 0 as _,
-        }
+        };
+        unsafe {
+            ffi::TPL_OpenTPLFromMemory(tpl.raw(), memory.as_mut_ptr() as *mut _, memory.len() as _)
+        };
+        tpl
     }
 
-    fn open_tpl_from_memory(&mut self, memory: &mut [u8]) {
-        unsafe { ffi::TPL_OpenTPLFromMemory(self.raw(), memory.as_mut_ptr() as *mut _, memory.len() as u32 )};
+    // Loads texture by id into texture
+    pub fn get_texture(&mut self, id: u32, texture: &mut gx::Texture) {
+        unsafe { ffi::TPL_GetTexture(self.raw(), id as _, texture.gxtexobj()) };
```

Comment:
You should be able to do `texture.into()` with the From impl?

Reply:
I tried to do that, but it didn't work.
```rust
error[E0277]: the trait bound `*mut _gx_texobj: From<&mut Texture<'_>>` is not satisfied
  --> src/tpl.rs:37:67
   |
37 |         unsafe { ffi::TPL_GetTexture(self.raw(), id as _, texture.into()) };
   |                                                                   ^^^^ the trait `From<&mut Texture<'_>>` is not implemented for `*mut _gx_texobj`
```

Reply:
Ohhh I see can you do a &*texture.into()?

Reply:
I cant resolve this :/